pkg/liqoctl/install/provider/provider_suite_test.go

```diff
@@ -19,9 +19,45 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/spf13/pflag"
+
+	"github.com/liqotech/liqo/pkg/liqoctl/install/provider"
 )
 
 func TestProvider(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Provider Suite")
 }
+
+func newFlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	flags.String("cluster-name", "", "")
+	flags.Bool("generate-name", false, "")
+	flags.String("reserved-subnets", "", "")
+	return flags
+}
+
+var _ = It("ValidateGenericCommandArguments", func() {
+	Describe("should accept a valid name", func() {
+		flags := newFlagSet()
+		clusterName := "test-cluster123"
+		Expect(flags.Set("cluster-name", clusterName)).To(Succeed())
+		provider := provider.GenericProvider{}
+		Expect(provider.ValidateGenericCommandArguments(flags)).To(Succeed())
+		Expect(provider.ClusterName).To(Equal(clusterName))
+	})
+	Describe("should not accept an invalid name", func() {
+		flags := newFlagSet()
+		clusterName := "Invalid cluster!"
+		Expect(flags.Set("cluster-name", clusterName)).To(Succeed())
+		provider := provider.GenericProvider{}
+		Expect(provider.ValidateGenericCommandArguments(flags)).ToNot(Succeed())
+	})
+	Describe("should generate a valid name if --generate-name is set", func() {
+		flags := newFlagSet()
+		Expect(flags.Set("generate-name", "true"))
+		provider := provider.GenericProvider{}
+		Expect(provider.ValidateGenericCommandArguments(flags)).To(Succeed())
+		Expect(provider.ClusterName).ToNot(BeEmpty())
+	})
+})
```

Comment:
This looks strange to me. Typically you have `It` blocks within `Describe` blocks, and not vice versa. (https://onsi.github.io/ginkgo/#structuring-your-specs)